azext_iot/common/central_functions.py

```diff
@@ -0,0 +1,149 @@
+    return token
+
+
+def validate_response_payload(response):
+    if response.status_code in [200, 201]:
+        return response.json()
+    exp = CloudError(response)
+    exp.request_id = response.headers.get("x-ms-request-id")
+    raise exp
+
+
+def get_iot_central_device_list(app_id, token):
+    import requests
+
+    host = get_app_host(app_id, token)["host"]
+    url = "https://{}/api/preview/devices".format(host)
+    response = requests.get(url, headers={"Authorization": "Bearer {}".format(token)})
+    return validate_response_payload(response)
+
+
+def get_app_host(app_id, token):
+    import requests
+
+    url = "https://apps.azureiotcentral.com/api/preview/applications/{}".format(app_id)
+    response = requests.get(url, headers={"Authorization": "Bearer {}".format(token)})
+    return validate_response_payload(response)
+
+
+def show_iot_central_provisioning_information(cmd, app_id, device_id):
+    aad_token = _get_aad_token(cmd, resource="https://apps.azureiotcentral.com")[
+        "accessToken"
+    ]
+    if device_id is None:
+        device_list = get_iot_central_device_list(app_id, aad_token)
+        print("Device Provisioning Summary:")
+        for item in device_list.get("value"):
+            print("\n")
+            data = {
+                i: item.get(i) for i in ["id", "displayName", "provisioned", "approved"]
+            }
+            print("{}".format(data))
+    else:
+        deviceCredentialData = get_iot_central_device_api_tokens(
+            cmd, app_id, device_id, aad_token
+        )
+        show_iot_central_device_provisioning_information(
+            deviceCredentialData["idScope"],
+            deviceCredentialData["symmetricKey"]["primaryKey"],
+            device_id,
+        )
+
+
+def get_iot_central_device_api_tokens(cmd, app_id, device_id, token):
+    import requests
+
+    host = get_app_host(app_id, token)["host"]
+    url = "https://{}/api/preview/devices/{}/credentials".format(host, device_id)
+    response = requests.get(url, headers={"Authorization": "Bearer {}".format(token)})
+    return validate_response_payload(response)
+
+
+def show_iot_central_device_provisioning_information(id_scope, primary_key, device_id):
+    provisioning_status = get_iot_central_device_provisioning_information(
+        id_scope, primary_key, device_id
+    )
+    print("provisioning information :\n")
+    for k, v in provisioning_status.items():
+        print("{} : {}\n".format(k, v))
+
+
+def get_iot_central_device_provisioning_information(id_scope, primary_key, device_id):
+    import requests
+
+    authToken = _get_dps_sas_auth_header(id_scope, device_id, primary_key)
+
+    url = "https://global.azure-devices-provisioning.net/{}/registrations/{}?api-version=2019-03-31".format(
+        id_scope, device_id
+    )
+    header_parameters = {}
+    header_parameters["Content-Type"] = "application/json"
```

Comment:
It would be nice adding the CLI extension user agent to the requests.

Reply:
Its available under `constants.USER_AGENT`